GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -298,41 +307,32 @@ private void OnUpdateStatusError(AsyncErrorEventArgs e)
 
         private void UpdatedStatusReceived(string updatedStatus)
         {
+            //Adjust the interval between updates. (This does not affect an update already scheculed).
+            _currUpdateInterval = Math.Max(MinUpdateInterval, 3 * (Environment.TickCount - _prevUpdateTime));
             _commandIsRunning = false;
 
             if (CurrentStatus != GitStatusMonitorState.Running)
                 return;
 
-            if (_statusIsUpToDate)
+            var allChangedFiles = GitCommandHelpers.GetAllChangedFilesFromString(Module, updatedStatus);
+            OnGitWorkingDirectoryStatusChanged(new GitWorkingDirectoryStatusEventArgs(allChangedFiles));
+            if (!_statusIsUpToDate)
             {
-                var allChangedFiles = GitCommandHelpers.GetAllChangedFilesFromString(Module, updatedStatus);
-                OnGitWorkingDirectoryStatusChanged(new GitWorkingDirectoryStatusEventArgs(allChangedFiles));
+                //Still not up-to-date, but present what received, GetAllChangedFilesCmd() is the heavy command
+                ScheduleNext(UpdateDelay);
             }
-            else
-            {
-                UpdateImmediately();
-            }
-        }
-
-        private void ScheduleNextJustInCaseUpdate()
-        {
-            _nextUpdateTime = Environment.TickCount + MaxUpdatePeriod;
         }
 
-        private void ScheduleDeferredUpdate()
+        private void ScheduleNext(int delay)
```

Comment:
I find this name confusing - the method isn't actually scheduling anything, it is calculating and updating the time of the next possible git command execution.
Please rename it to something like `CalculateNextUpdateTime`